flink-runtime/src/test/java/org/apache/flink/streaming/util/KeyedTwoInputStreamOperatorTestHarness.java

```diff
@@ -64,8 +67,24 @@ public KeyedTwoInputStreamOperatorTestHarness(
     }
 
     public int numKeyedStateEntries() {
-        AbstractStreamOperator<?> abstractStreamOperator = (AbstractStreamOperator<?>) operator;
-        KeyedStateBackend<Object> keyedStateBackend = abstractStreamOperator.getKeyedStateBackend();
+        KeyedStateBackend<?> keyedStateBackend;
+        if (operator instanceof AbstractAsyncStateStreamOperator) {
```

Comment:
as reflection is quite costly and the operator is an instance variable, can we organize the code so the reflection is done once and caches rather than on every access.

I suggest either working out what type of operator we have in the constructor and stash in an instance variable; maybe using an enum for the types of operators